// This file is part of KWIVER, and is distributed under the
// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.

syntax = "proto2";

package kwiver.protobuf;

import "geo_point.proto";
import "geo_polygon.proto";

message metadata {
  message metadata_item {

    // Metadata tag value
    required int32  metadata_tag = 1;

    // Only one of these should be present. The "oneof" feature can be
    // used when it becomes available.
    optional string string_value = 2;
    optional int64  int_value = 3;
    optional double double_value = 4;

    // It is not ideal to add specific data types to this "union"
    // but serializing them any other way is no better.
    optional geo_point geo_point_value = 5;
    optional geo_polygon geo_polygon_value = 6;
  }

  repeated metadata_item items = 6;

}

// Vector of metadata items
message metadata_vector {

  repeated metadata collection = 1;

}
